packages/vc-verification/src/revocation-verification.ts

```diff
@@ -0,0 +1,107 @@
+import { addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { verifyCredential } from 'didkit-wasm-node';
+import { AuthorityResolver, CredentialResolver } from '.';
+import { StatusPurpose, StatusListCredential } from './models';
+import { VCIssuerVerification } from './issuer-verification-vc';
+
+/**
+ * A class to validate if the revocation is valid or not
+ */
+export class RevocationVerification {
+  private _authorityResolver: AuthorityResolver;
+  private _credentialResolver: CredentialResolver;
+  private _issuerVerification: VCIssuerVerification;
+
+  /**
+   *
+   * @param revokerDefResolver
+   * @param credentialResolver
+   */
+  constructor(
+    authorityResolver: AuthorityResolver,
+    credentialResolver: CredentialResolver
+  ) {
+    this._authorityResolver = authorityResolver;
+    this._credentialResolver = credentialResolver;
+    this._issuerVerification = new VCIssuerVerification(
+      credentialResolver,
+      authorityResolver
+    );
+  }
+
+  /**
+   * Verify revocation for the provided credential or namespace and revoker
+   * @param credential
+   * @param namespace
+   * @param revokerDID
+   * @returns
+   */
+  async verifyRevocation(namespace: string, credential: StatusListCredential) {
+    const revokerDID = credential.issuer;
+    if (
+      !this.isStatusPurposeRevocation(
+        credential.credentialSubject.statusPurpose
+      )
+    ) {
+      return false;
+    }
+    return await this.verifyRevokerAuthority(namespace, revokerDID);
+  }
+
+  /**
+   *
+   * @param namespace for which the revoker's authority is to be verified
+   * @param revokerDID whose authority needs to be verified
+   * @returns
+   */
+  async verifyRevokerAuthority(
+    namespace: string,
+    revokerDID: string
+  ): Promise<boolean> {
+    const revokers = await this._authorityResolver.getRevokerDefinition(
+      namespace
+    );
+    if (revokers && revokers.did) {
+      for (let i = 0; i < revokers.did.length; i++) {
+        const revokerAddr = addressOf(revokerDID);
+        const roleRevokerAddr = addressOf(revokers.did[i]);
+        if (roleRevokerAddr.toUpperCase() === revokerAddr.toUpperCase()) {
+          return true;
+        }
+      }
+    }
+    let isCredentialVerified = false;
+    let isCredentialIssuerVerified = false;
+    if (revokers && revokers.roleName) {
```

Comment:
Similar to previous comment
```suggestion
    if (revokers?.roleName) {
```